backend/kernelCI_app/views/hardwareDetailsView.py

```diff
@@ -224,7 +250,12 @@ def sanitize_records(self, records, selected_trees, filters):
 
         for r in records:
             current_tree = get_current_selected_tree(r, selected_trees)
-            if not current_tree or not is_record_selected(r, current_tree):
+
+            is_tree_selected = current_tree and is_record_selected(r, current_tree)
+            has_global_filter = len(filters_map['global']) == 0
+            pass_in_global_filters = has_global_filter or self.pass_build_filter(r, filters_map['global'])
```

Comment:
these names are a bit confusing, `has_global_filter` is True if the length of the filters is 0? Also, should `pass_in_global_filters` be using `or` in its comparison?